docs/Technical-Documentation/Architecture-Decision-Record/017-elastisearch.md

```diff
@@ -0,0 +1,29 @@
+# 17. Elastisearch Architecture
+
+Date: 2021-01-12 (yyyy-mm-dd)
+
+## Status
+
+Accepted
+
+## Context
+
+To modernize the eventual reporting stack for TDP, we have a need for a usable data backend.
+
+## Decision
+
+We will use an Elastisearch and Kibana stack for their modern feature set and scalability with large data sets. The Elastisearch/Kibana cluster will be an application hosted in Cloud.gov with the rest of our TDP application so we can leverage all the work surrounding authentication, security, and data compliance.
+
+## Consequences
+
+### Pros
+ - Elastisearch is fast becoming the industry standard if not already the standard for large data sets and getting their feature sets for customizable queries, scalability, and advanced storage methods.
+ - Kibana frontend will be very user friendly compared to home rolling our own reporting interface.
+
+
+### Cons
+By divorcing the database(s) of TANF data from our application's PostGre SQL database, we have introduced complexity in the architecture both for accessing this data in Kibana but also in the user hand-off from our frontend to the Kibana UI.
+
+## Notes
```

Comment:
suggest linking these 2 additional resources:

- [tdp_es_demo.md](https://gist.github.com/jtwillis92/a6840a412676fc2d3f58c0dccbf10da1)
- [generate_mock_tanf_data.py](https://gist.github.com/jtwillis92/50dd8f578b78ff1c4a9343a28958d338)

Reply:
JW noted [here](https://github.com/raft-tech/TANF-app/issues/1289#issuecomment-914629307) that it might be useful for unit testing later.

Reply:
Yes, I think the script itself is valuable but I don't think it is pertinent to include in the ADR. Should we reference it in the actual Elasticsearch tickets? Absolutely. I hope this clarifies why I'm leaving it out.